public/class-monk-public.php

```diff
@@ -89,7 +89,7 @@ public function monk_public_posts_filter( $query ) {
 			return;
 		}
 
-		global $monk_languages;
+		$monk_languages = get_transient( 'monk_languages' );
```

Comment:
Change all occurrences of `get_transient( 'monk_languages' );` to `monk_get_available_languages();`